x-pack/plugin/inference/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/application/AuthorizationTaskExecutorUpgradeIT.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.upgrades.ParameterizedRollingUpgradeTestCase;
+import org.elasticsearch.xpack.inference.services.elastic.authorization.AuthorizationPoller;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.inference.InferenceBaseRestTest.assertStatusOkOrCreated;
+import static org.elasticsearch.xpack.inference.services.elastic.ElasticInferenceServiceSettings.ELASTIC_INFERENCE_SERVICE_URL;
+import static org.elasticsearch.xpack.inference.services.elastic.ElasticInferenceServiceSettings.PERIODIC_AUTHORIZATION_ENABLED;
+import static org.hamcrest.Matchers.is;
+
+public class AuthorizationTaskExecutorUpgradeIT extends ParameterizedRollingUpgradeTestCase {
+
+    private static final String BEFORE_AUTHORIZATION_TASK_FEATURE = "gte_v9.2.0";
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .version(getOldClusterVersion(), isOldClusterDetachedVersion())
+        .nodes(NODE_NUM)
+        .setting("xpack.security.enabled", "false")
+        .setting("xpack.license.self_generated.type", "trial")
+        .setting(PERIODIC_AUTHORIZATION_ENABLED.getKey(), "false")
+        // We need a url set for the authorization task to be created, but we don't actually care if we get a valid response
+        // just that the task will be created upon upgrade
+        .setting(ELASTIC_INFERENCE_SERVICE_URL.getKey(), "http://localhost:12345")
+        .build();
+
+    private static final String GET_METHOD = "GET";
+
+    public AuthorizationTaskExecutorUpgradeIT(@Name("upgradedNodes") int upgradedNodes) {
+        super(upgradedNodes);
+    }
+
+    @Override
+    protected ElasticsearchCluster getUpgradeCluster() {
+        return cluster;
+    }
+
+    public void testUpgradeAuthorizationTaskExecutor() throws Exception {
+        assumeTrue("Authorization Polling Task supported", oldClusterHasFeature(BEFORE_AUTHORIZATION_TASK_FEATURE));
+
+        // If the old cluster already has the feature for the authorization polling task, the task should already exist
+        // We only want to test when upgrading from a version that does not have the task
+        if (oldClusterHasFeature(BEFORE_AUTHORIZATION_TASK_FEATURE) == false) {
```

Comment:
The contents of this block are unreachable, because the test will exit on the `assumeTrue()` call above before reaching here if `oldClusterHasFeature(BEFORE_AUTHORIZATION_TASK_FEATURE)` returns false.